### PR TITLE
fix dashboard customer_report -> reciept_report

### PR DESCRIPTION
**Pull request by Npwskp**
fix follow design
still discuss with start time
image

### CODE REVIEW

File: features/step-definitions/session.ts [hunk 1]
```diff
@@ -20,10 +20,10 @@ export class SessionStepDefination {
     for (const session of sessions) {
       const doc = await this.sessionModel.create({
         table: new Types.ObjectId(session.table),
-        user: session.user ?? null,
+        user: session.user ? session.user : null,
         _id: new Types.ObjectId(session._id),
         finished_at: session.finished_at ? new Date(session.finished_at) : null,
-        point: session.point ?? 0,
+        point: session.point ? session.user : 0,
```
Comment (lazykern) [inline features/step-definitions/session.ts]: dafuq

File: src/schema/session.schema.ts [hunk 2]
```diff
@@ -31,7 +31,12 @@ export class SessionSchema {
   })
   finished_at: Date;
 
-  @Prop({ default: null, ref: () => UserSchema })
+  @Prop({
+    default: null,
+    ref: () => UserSchema,
+    required: false,
```
Comment (lazykern) [inline src/schema/session.schema.ts]: revert this back to how it was

File: src/pipes/mongo-datestart.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+} from '@nestjs/common';
+
+@Injectable()
+export class ParseMongoDateStartPipe implements PipeTransform {
+  transform(value: number) {
+    try {
+      const hour = 86400;
+      const new_value = value - (value % hour);
+      const mul = 1000;
+      return new Date(new_value * mul);
```
Comment (lazykern) [inline src/pipes/mongo-datestart.pipe.ts]: This works but it seems so weird. You might wanna use `setHours` to make it look cleaner. Anyway, it's optional.

File: src/dashboard/dto/getAllReceiptAmount.dto.ts [hunk 4]
```diff
@@ -1,21 +1,21 @@
...
     description: 'Total old registered users',
     type: Number,
   })
-  old_user: number;
+  receipt_user: number;
 
   @ApiProperty({
     description: 'Total new registered users',
     type: Number,
   })
```
Comment (lazykern) [inline src/dashboard/dto/getAllReceiptAmount.dto.ts]: Update the description

File: src/dashboard/dashboard.controller.ts [hunk 5]
```diff
@@ -29,18 +22,25 @@ export class DashboardController {
...
   })
   @ApiOperation({
-    summary: 'Get total registered users',
+    summary: 'Get total recirpt amount',
+  })
+  @ApiQuery({
+    name: 'date',
+    type: Number,
+    required: true,
+    description: 'Date (UnixTimeStamp)',
```
Comment (lazykern) [inline src/dashboard/dashboard.controller.ts]: (Optional) please also provide that the timestamp muse be in seconds

File: src/dashboard/dashboard.controller.ts [hunk 6]
```diff
@@ -29,18 +22,25 @@ export class DashboardController {
 
   @ApiResponse({
     status: HttpStatus.OK,
-    type: () => GetUserAmountDto,
-    description: 'Total registered users',
+    type: () => GetReceiptAmountDto,
+    description: 'Total recirpt amount',
```
Comment (lazykern) [inline src/dashboard/dashboard.controller.ts]: typo

File: features/step-definitions/session.ts (hunk 1 above)
Comment (lazykern) [reply features/step-definitions/session.ts]: it's the point, not user, ma boi

Comment (Npwskp) [reply features/step-definitions/session.ts]: bruh

File: src/pipes/mongo-datestart.pipe.ts (hunk 3 above)
Comment (Npwskp) [reply src/pipes/mongo-datestart.pipe.ts]: i go look for sethour() it set just hour if i want 00:00:00 sethour(0),setminute(0),setsecond(0) right?

File: src/schema/session.schema.ts (hunk 2 above)
Comment (Npwskp) [reply src/schema/session.schema.ts]: no

File: src/pipes/mongo-datestart.pipe.ts (hunk 3 above)
Comment (Npwskp) [reply src/pipes/mongo-datestart.pipe.ts]: ok i get it

File: src/dashboard/dashboard.controller.ts (hunk 6 above)
Comment (Npwskp) [reply src/dashboard/dashboard.controller.ts]: k

File: src/schema/session.schema.ts (hunk 2 above)
Comment (lazykern) [reply src/schema/session.schema.ts]: fuck you

File: src/dashboard/dashboard.controller.ts (hunk 5 above)
Comment (Npwskp) [reply src/dashboard/dashboard.controller.ts]: discuss w/ golf he said no

Comment (Npwskp) [reply src/dashboard/dashboard.controller.ts]: frontend not gonna be dumb if it false its their false

File: src/pipes/mongo-datestart.pipe.ts (hunk 3 above)
Comment (Npwskp) [reply src/pipes/mongo-datestart.pipe.ts]: discuss golf la he said front gonna sent 00.00.00

**Review by lazykern — DISMISSED** [review]

**Review by lazykern — APPROVED** [review]

**Pull request merged**